tests/sync/test_assertions.py

```diff
@@ -158,6 +158,81 @@ def test_assertions_locator_to_have_js_property(page: Page, server: Server) -> N
     )
 
 
+def test_pass_string(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = 'string'")
+    locator = page.locator("div")
+    expect(locator).to_have_js_property("foo", "string")
+
+
+def test_fail_string(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = 'string'")
+    locator = page.locator("div")
+    with pytest.raises(AssertionError):
+        expect(locator).to_have_js_property("foo", "error", timeout=500)
+
+
+def test_pass_number(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = 2021")
+    locator = page.locator("div")
+    expect(locator).to_have_js_property("foo", 2021)
+
+
+def test_fail_number(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = 2021")
+    locator = page.locator("div")
+    with pytest.raises(AssertionError):
+        expect(locator).to_have_js_property("foo", 1, timeout=500)
+
+
+def test_pass_boolean(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = true")
+    locator = page.locator("div")
+    expect(locator).to_have_js_property("foo", True)
+
+
+def test_fail_boolean(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = false")
+    locator = page.locator("div")
+    with pytest.raises(AssertionError):
+        expect(locator).to_have_js_property("foo", True, timeout=500)
+
+
+def test_pass_boolean_2(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = false")
+    locator = page.locator("div")
+    expect(locator).to_have_js_property("foo", False)
+
+
+def test_fail_boolean_2(page: Page) -> None:
+    page.set_content("<div></div>")
+    page.eval_on_selector("div", "e => e.foo = false")
+    locator = page.locator("div")
+    with pytest.raises(AssertionError):
+        expect(locator).to_have_js_property("foo", True, timeout=500)
+
+
+def test_pass_undefined(page: Page) -> None:
+    page.set_content("<div></div>")
+    locator = page.locator("div")
+    expect(locator).to_have_js_property(
+        "foo", None
```

Comment:
What should this test be in Python? In the Node version, this is `undefined` instead of `None`.

As is, this fails like:

```
    async def test_pass_undefined(page: Page) -> None:
            await page.set_content('<div></div>')
            locator = page.locator('div')
>           await expect(locator).to_have_js_property('foo', None) # Python does not have an undefined
E           AssertionError: Locator expected to have JS Property
E           Actual value: None 
E           Call log:
E           locator_assertions.to_have_js_property with timeout 5000ms
E           waiting for selector "div"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
E             selector resolved to <div></div>
E             unexpected value "undefined"
```

Reply:
lets talk today in the standup about it.

Reply:
Conclusion: 

- null in Python/.NET should mean undefined on the server side.
- ignore the null case for now, we could add some server side change to handle both cases when undefined is passed in the future.

i.e.

- if you pass null it should assert undefined in the browser
- follow up accepting null and undefined in a separate patch post 1.23